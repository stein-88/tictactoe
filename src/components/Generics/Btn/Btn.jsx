import React from 'react'
import s from './Btn.scss'

const Btn = ({
 onClick, text, disabledbtn, customClass, id,
}) => {
	if (!text || !onClick) return null
	return (
		<span
			onClick={(e) => {
				if (!disabledbtn) return onClick(e)
				return null
			}}
			id={id}
			data-test={id}
			key={id}
			aria-hidden="true"
			className={`${!disabledbtn ? '' : s.btnoff} ${customClass || s.btnstyle}`.trim()}
		>
			{text}
		</span>
	)
}

export default Btn
