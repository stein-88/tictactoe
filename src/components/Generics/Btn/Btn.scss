.btnstyle {
  border: 3px solid #0000ff52;
  color: white;
  padding: 4px 5px;
  border-radius: 8px;
  display: inline-block;
  cursor: pointer;
  background-color: #007bff;
  margin-right: 7px;

  &.btnoff {
    background-color: lighten($color: black, $amount: 40) !important;
    cursor: none !important;
  }
}
