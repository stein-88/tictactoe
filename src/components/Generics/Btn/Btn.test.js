import React from 'react'
import renderer from 'react-test-renderer'
import Btn from './Btn'

describe('Test on Btn Component', () => {
    test('Should not render without props', () => {
        const compBtn = renderer.create(<Btn />)
        const jsonBtn = compBtn.toJSON()
        expect(jsonBtn).toBeNull()
    })
    test('Should not render with onClick only', () => {
        const compBtn = renderer.create(<Btn onClick={() => null} />)
        const jsonBtn = compBtn.toJSON()
        expect(jsonBtn).toBeNull()
    })
    test('Should not render with text only', () => {
        const compBtn = renderer.create(<Btn text="botaoo" />)
        const jsonBtn = compBtn.toJSON()
        expect(jsonBtn).toBeNull()
    })
    test('Should render with text and onClick', () => {
        const compBtn = renderer.create(<Btn text="botaoo" onClick={() => null} />)
        const jsonBtn = compBtn.toJSON()
        expect(jsonBtn).toMatchSnapshot()
    })
    test('Should change customClass', () => {
        const mockFunc = jest.fn()
        const compBtn = renderer.create(<Btn text="botaoo" customClass="abobrinha" onClick={mockFunc} />)
        const result = compBtn.root.findAllByProps({ className: 'abobrinha' })
        result[0].props.onClick()
        expect(mockFunc).toHaveBeenCalled()
        expect(result).toHaveLength(1)
    })
    test('Should add btnoff to className when we have disablebtn', () => {
        const mockFunc = jest.fn()
        const compBtn = renderer.create(<Btn text="botaoo" disabledbtn={true} onClick={mockFunc} />)
        const expectResult = compBtn.root.findAllByProps({ className: 'btnoff btnstyle' })
        expectResult[0].props.onClick()
        expect(mockFunc).not.toHaveBeenCalled()
        expect(expectResult).toHaveLength(1)
    })
    test('Should have an ID', () => {
        const compBtn = renderer.create(<Btn text="botaoo" id="tecladoGamerTeclaBaixa" onClick={() => null} />)
        const expectResult = compBtn.root.findAllByProps({ ['data-test']: "tecladoGamerTeclaBaixa" })
        expect(expectResult).toHaveLength(1)
    })
})