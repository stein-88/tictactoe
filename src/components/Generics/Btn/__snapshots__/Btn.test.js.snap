// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on Btn Component Should render with text and onClick 1`] = `
<span
  aria-hidden="true"
  className="btnstyle"
  onClick={[Function]}
>
  botaoo
</span>
`;
